include/boost/geometry/strategies/geographic/closest_points_pt_seg.hpp

```diff
@@ -0,0 +1,121 @@
+#include <boost/geometry/strategies/spherical/distance_haversine.hpp>
+#include <boost/geometry/strategies/spherical/point_in_point.hpp>
+#include <boost/geometry/strategies/geographic/azimuth.hpp>
+#include <boost/geometry/strategies/geographic/distance.hpp>
+#include <boost/geometry/strategies/geographic/distance_cross_track.hpp>
+#include <boost/geometry/strategies/geographic/parameters.hpp>
+#include <boost/geometry/strategies/geographic/intersection.hpp>
+
+#include <boost/geometry/util/math.hpp>
+#include <boost/geometry/util/select_calculation_type.hpp>
+#include <boost/geometry/util/normalize_spheroidal_coordinates.hpp>
+
+#include <boost/geometry/formulas/mean_radius.hpp>
+#include <boost/geometry/formulas/result_direct.hpp>
+#include <boost/geometry/formulas/spherical.hpp>
+#include <boost/geometry/formulas/vincenty_direct.hpp>
+
+#ifdef BOOST_GEOMETRY_DEBUG_GEOGRAPHIC_CROSS_TRACK
+#include <boost/geometry/io/dsv/write.hpp>
+#endif
+
+#ifndef BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS
+#define BOOST_GEOMETRY_DETAIL_POINT_SEGMENT_DISTANCE_MAX_STEPS 100
+#endif
+
+#ifdef BOOST_GEOMETRY_DEBUG_GEOGRAPHIC_CROSS_TRACK
+#include <iostream>
+#endif
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace closest_points
+{
+
+template
+<
+    typename FormulaPolicy = geometry::strategy::andoyer,
+    typename Spheroid = srs::spheroid<double>,
+    typename CalculationType = void
+>
+class geographic_cross_track
+    : public distance::detail::geographic_cross_track
+        <
+            FormulaPolicy,
+            Spheroid,
+            CalculationType,
+            false,
+            true
+        >
+{
+    using base_t = distance::detail::geographic_cross_track
+        <
+            FormulaPolicy,
+            Spheroid,
+            CalculationType,
+            false,
+            true
+        >;
+
+public :
+    explicit geographic_cross_track(Spheroid const& spheroid = Spheroid())
+        : base_t(spheroid)
+        {}
+
+        template <typename Point, typename PointOfSegment>
+        auto apply(Point const& p,
+                   PointOfSegment const& sp1,
+                   PointOfSegment const& sp2) const
+        {
+            auto result = base_t::apply(get_as_radian<0>(sp1), get_as_radian<1>(sp1),
+                                        get_as_radian<0>(sp2), get_as_radian<1>(sp2),
+                                        get_as_radian<0>(p), get_as_radian<1>(p),
+                                        base_t::m_spheroid);
+
+            model::point
+                <
+                    typename base_t::template return_type<Point, PointOfSegment>::type,
+                    dimension<PointOfSegment>::value,
+                    typename coordinate_system<PointOfSegment>::type
+                > cp;
+            
+            geometry::set_from_radian<0>(cp, result.lon);
+            geometry::set_from_radian<1>(cp, result.lat);
+
+            return cp;
```

Comment:
So the result is a point using the coordinate type being the result of distance? If that's the case here then for consistency it has to be the case in cartesian CS too. So technically it's possible to pass points with integral coordinates and get a point with double coordinates as the output. Is that what we want?

Reply:
No, this should be just like in spherical. Changed.

Reply:
Rethinking of it, we probably should always promote to floating point since even in the case of input integral coordinates the closest point on the segment is possibly not be able to be represented by integral coordinates. This should happen in all CS.

Reply:
Yes, it makes sense. I don't know if this is correct since a user will typically pass the same point type as the in the input and output. Still internally an FP type can be used for calculation purposes even if at the end it will be converted to user-defined type.

If you take the coordinate type from the distance result in practice it is going to be FP in all coordinate systems.

Reply:
OK, I left it as is and added a unit test with integral coordinates.